lighthouse-core/audits/audit.js

```diff
@@ -57,6 +57,60 @@ class Audit {
   }
 
   /**
+   * @param  {!Audit.Headings} headings
+   * @return {*}
+   */
+  static makeV1TableHeadings(headings) {
+    const tableHeadings = {};
+    headings.forEach(heading => tableHeadings[heading.key] = heading.text);
+    return tableHeadings;
+  }
+
+  /**
+   * @param  {!Audit.Headings} headings
+   * @param  {!Object} results
+   * @return {!Array<!DetailsRenderer.DetailsJSON>}
+   */
+  static makeV2TableRows(headings, results) {
+    const tableRows = results.map(item => {
+      return headings.map(heading => {
+        return {
+          type: heading.itemType,
+          text: item[heading.key]
+        };
+      });
+    });
+    return tableRows;
+  }
+
+  /**
+   * @param  {!Audit.Headings} headings
+   * @return {!Array<!DetailsRenderer.DetailsJSON>}
```

Comment:
this is fine for now, but it won't actually work if we were to turn on type checking on this file. Closure won't be able to see the typedef since it's in another module. We'll have to think of how we want to do that (move report typedefs to their own file?)